packages/account/src/Components/currency-selector/__tests__/currency-selector.spec.tsx

```diff
@@ -241,78 +60,302 @@ describe('<CurrencySelector/>', () => {
+                            AUD: 2,
+                        },
+                        limits: {
+                            max: 5001.78,
+                            min: 1,
+                        },
+                        limits_dxtrade: {
+                            max: 2500.89,
+                            min: 0.01,
+                        },
+                        limits_mt5: {
+                            max: 15005.33,
+                            min: 0.01,
+                        },
+                    },
+                    type: 'crypto',
+                },
+            ],
+        },
+        ui: {
+            real_account_signup: {
+                active_modal_index: -1,
+                previous_currency: '',
+                current_currency: '',
+                success_message: '',
+                error_message: '',
+                error_code: '2',
+            },
+        },
+    });
+
+    const renderComponent = ({ props = mock_props, store_config = store }) => {
+        return render(
+            <StoreProvider store={store_config}>
+                <CurrencySelector {...props} />
+            </StoreProvider>
+        );
+    };
+
+    it('should render Currency selector', () => {
+        renderComponent({});
 
-    it('should render currencyselector', () => {
-        render(<CurrencySelector {...props} />);
         expect(screen.getByTestId('currency_selector_form')).toBeInTheDocument();
     });
 
     it('should render Fiat currencies and submit the form', async () => {
-        render(<CurrencySelector {...props} />);
+        renderComponent({});
+
         runCommonTests(fiat_msg);
         fireEvent.click(screen.getByRole('button', { name: /next/i }));
         await waitFor(() => {
-            expect(props.onSubmit).toHaveBeenCalled();
-            expect(props.onSubmit).toHaveBeenCalledWith(
+            expect(mock_props.onSubmit).toHaveBeenCalled();
+            expect(mock_props.onSubmit).toHaveBeenCalledWith(
                 0,
                 { currency: 'USD' },
                 expect.any(Function),
-                props.goToNextStep
+                mock_props.goToNextStep
             );
         });
     });
 
     it('should disable fiat if user already have a fiat ', () => {
-        const new_props = {
-            ...props,
-            accounts: {
-                VRTC90000010: {
-                    account_type: 'trading',
-                    currency: 'USD',
-                    is_disabled: 0,
-                    is_virtual: 1,
-                    landing_company_shortcode: 'svg',
-                    trading: {},
-                    token: '',
-                    email: '',
-                    session_start: 1651059038,
-                    excluded_until: '',
-                    landing_company_name: 'svg',
-                    residence: 'es',
-                    balance: 10000,
-                    accepted_bch: 0,
+        const new_store = {
```

Comment:
I feel it is an overkill in Testcase

Reply:
types are not inferred, when you just use ‘…’ operator.
we may skip types here, just we should make sure, that we add new properties correctly to the mocked data, in order to avoid TS errors.